common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -454,7 +455,15 @@ def parse_query_params(strategy, response, *args, **kwargs):
     # We simply assume 'login' in that case.
     auth_entry = strategy.request.session.get(AUTH_ENTRY_KEY, AUTH_ENTRY_LOGIN)
     if auth_entry not in _AUTH_ENTRY_CHOICES:
+        if not auth_entry and ENABLE_OKTA_AUTH_FIX.is_enabled():
+            # This change is for get okta working with the pipeline.
+            # In some clients, the pipeline starts in okta, so we don't have
+            # this in the field in the session, because there there is no
+            # session yet. So we return login, but the standard workflow still
+            # works.
+            return {'auth_entry': 'login'}
```

Comment:
Could you add a unit test for this code, maybe like [`third_party_auth.tests.specs.base.test_pipeline_assumes_login_if_auth_entry_missing`](https://github.com/edx/edx-platform/blob/c3571a5fb9f6d85665c5c58dd723b760108bc6d1/common/djangoapps/third_party_auth/tests/specs/base.py#L908-L911)?  Since the default cases are already verified, you just need to verify that the login view is shown when the added waffle switch is enabled.